activerecord/lib/active_record.rb

```diff
@@ -464,6 +464,15 @@ def self.marshalling_format_version=(value)
     Marshalling.format_version = value
   end
 
+  ##
+  # :singleton-method:
+  # Sets the database connection adapter to be used when the adapter <tt>mysql</tt> is specified.
+  # Accepted values are <tt>:mysql2</tt>, <tt>:trilogy</tt>, and <tt>nil</tt> which does not register
+  # any <tt>mysql</tt> adapter alias.
+  # Defaults to <tt>nil</tt>.
+  singleton_class.attr_accessor :mysql_adapter
+  self.mysql_adapter = nil
```

Comment:
```suggestion
  self.mysql_adapter = "trilogy"
```

Reply:
I will make this the default. We solved the load order problem so this is possible now.

But, I do believe that as you pointed out `caching_sha2_password` support https://github.com/trilogy-libraries/trilogy/issues/26 is a blocker and should be done for Rails 7.2.

Reply:
I'm fine with leaving `mysql2` the default in this PR, and then have a small followup to switch to trilogy that we'll merge as soon as this blocker is solved upstrean.